mqtt/mqttd/src/broker/mod.rs

```diff
@@ -41,84 +47,82 @@ where
     let server_join_handle =
         tokio::spawn(bootstrap::start_server(settings, broker, shutdown_signal));
 
-    // start sidecars
-    // TODO REVIEW: it is fine to blow up here
-    //              a failure here would indicate either bridge or command handler init failed
-    //              broker doesn't open ports until these are both initialized
+    // start sidecars if they should run
+    // if not wait for server shutdown
     let state;
-    if let Some((sidecars_shutdown, sidecar_join_handles)) =
+    if let Some(sidecar_manager) =
         bootstrap::start_sidecars(broker_handle.clone(), listener_settings).await?
     {
-        // combine future for all sidecars
         // wait on future for sidecars or broker
         // if one of them exits then shut the other down
-        // TODO REVIEW: we are only blowing up if either:
-        //              1 - we can't stop the server and can't get the needed state
-        //              2 - we can't signal server or sidecars to shutdown
-        let sidecars_fut = select_all(sidecar_join_handles);
+        let sidecar_shutdown_handle = sidecar_manager.shutdown_handle();
+        let sidecars_fut = sidecar_manager.wait_for_shutdown();
```

Comment:
We discussed offline how we also need to expose some future for sidecars that we can select on so that if any sidecar stops, we can shut the server down. I have added comments to the SidecarManager.